aiida/backends/sqlalchemy/querybuilder_sqla.py

```diff
@@ -605,6 +604,8 @@ def iterdict(self, query, batch_size, tag_to_projected_entity_dict):
                             }
                             for tag, projected_entities_dict in tag_to_projected_entity_dict.items()
                         }
+            else:
+                raise Exception("Got an empty dictionary")
```

Comment:
Apart for the test that is failing (that I guess it's simple to fix), could you replace generic raised Exceptions with appropriate ones (here and elsewhere)? Either defined in aiida.common.exceptions, or most probably it's enough to use the standard Python ones (e.g. here I imagine a `ValueError`, other times it might be a `TypeError` or similar).

Reply:
OK for the exceptions, I put in more appropriate (b037028)
I can't reproduce the failing test locally though.